src/mawpy/workflows/ao_ic_usd.py

```diff
@@ -0,0 +1,51 @@
+# importing the modules
+import datetime
+import logging
+
+import multiprocessing
+
+import pandas as pd
+
+from mawpy.steps.address_oscillation import address_oscillation
+from mawpy.steps.incremental_clustering import incremental_clustering
+from mawpy.steps.update_stay_duration import update_stay_duration
+import os
+
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("input_file", help="the CSV file to read the input from")
+parser.add_argument("output_file", help="the CSV file to write the output to.")
+parser.add_argument('--spatial_constraint', default=1.0, type=float, required=False)
+parser.add_argument('--duration_constraint_1', default=0, type=float, required=False)
+parser.add_argument('--duration_constraint_2', default=300, type=float, required=False)
+parser.add_argument('--duration_constraint_3', default=300, type=float, required=False)
+
+logger = logging.getLogger(__name__)
+
+
+def ao_ic_usd(
+    input_file: str,
+    output_file: str,
+    spatial_constraint: float,
+    duration_constraint1: float,
+    duration_constraint2: float,
+    duration_constraint3: float
+) -> pd.DataFrame:
+    df_output_ao = address_oscillation(output_file, duration_constraint1, input_file=input_file)
+    df_output_ic = incremental_clustering(output_file, spatial_constraint, duration_constraint2, input_df=df_output_ao)
+    df_output_final = update_stay_duration(output_file, duration_constraint3, input_df=df_output_ic)
+    return df_output_final
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+    multiprocessing.freeze_support() # TODO: do we require this? most probably NOT.
+    st = datetime.datetime.now()
+    ao_ic_usd(args.input_file, "output_file.csv", args.spatial_constraint, args.duration_constraint_1,
+              args.duration_constraint_2, args.duration_constraint_3)
+    en = datetime.datetime.now()
+    logger.info(f"Total Time taken for execution: {en - st}")
+    current_filename = "output_file.csv"
```

Comment:
use temp file instead , same below

Reply:
The file here is not temporary. The workflow is such that the file is renamed to its final name only when all the steps of the workflow are completed.

In short, we use the same file and rename it later to the value passed in the command line argument.
I believe its to distinguish between if the workflow was completed or if the code broke during execution.

Reply:
I see. In that case, two suggestions:

- Make the file name a bit less common. I could easily see someone passing `--output_file output_file.csv` and getting an error or at least not the desired behavior. Maybe something like `"_ao_ic_usd_temp.csv"`. That would also help with file names not overlapping/overwriting between different workflows.
- Define these file name as a constant, rather than repeating the literal